lib/synapse/config_generator/haproxy.rb

```diff
@@ -961,17 +973,28 @@ def validate_haproxy_stanza(stanza, stanza_type, service_name)
 
     # generates an individual stanza for a particular watcher
     def generate_frontend_stanza(watcher, config)
-      unless watcher.haproxy.has_key?("port")
+      unless watcher.generator_config['haproxy'].has_key?("port")
```

Comment:
can we declare a `watcher_opts` var so we don't have to keep calling `watcher.generator_config['haproxy']`? also, should we be hard-coding `'haproxy'`? i think we should use `name`:

```ruby
watcher_opts = watcher.generator_config[name]
```